NumberView.cpp

```diff
@@ -30,6 +37,14 @@ NumberView::SetNumber(uint32 number)
 void
 NumberView::Draw(BRect r)
 {
+	// Update squares everytime a window is resized
+	SetExplicitMinSize(BSize(squareSize, squareSize));
+	SetExplicitMaxSize(BSize(squareSize, squareSize));
+	SetExplicitPreferredSize(BSize(squareSize, squareSize));
+	BFont *font = new BFont(be_bold_font);
+	font->SetSize(squareSize/2);
+	SetFont(font);
```

Comment:
This block looks very similar to the constructor. Try to not repeat yourself and factor it out in a separate method, called both here and from the constructor.
Also, it looks inefficient to do this each time Draw is called. Why not do it just once in SetScale instead?